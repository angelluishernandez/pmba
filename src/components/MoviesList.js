import React from "react";
import MovieItem from "./MovieItem";

const MoviesList = ({ movies }) => {
	const sortedMovies = movies.sort(
		(movieA, movieB) => movieA.watched - movieB.watched
	);

	return (
		<div>
			{sortedMovies.map((movie, index) => (
				<MovieItem movie={movie} key={index} />
			))}
		</div>
	);
};

export default MoviesList;
