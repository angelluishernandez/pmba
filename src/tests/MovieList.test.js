import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJSON from "enzyme-to-json";
import React from "react";

import MoviesList from "../components/MoviesList";

configure({ adapter: new Adapter() });

test("should render MoviesList correctly", () => {
	const movies = [];
	const wrapper = shallow(<MoviesList movies={movies} />);
	expect(toJSON(wrapper)).toMatchSnapshot();
});

test("should render MovieList with empty list message", () => {
	const wrapper = shallow(<MoviesList movies={[]} />);
	expect(toJSON(wrapper)).toMatchSnapshot();
});
