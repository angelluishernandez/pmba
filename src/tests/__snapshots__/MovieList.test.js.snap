// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render MovieList with empty list message 1`] = `<div />`;

exports[`should render MoviesList correctly 1`] = `
<div>
  <Connect(MovieItem)
    key="0"
    movie={
      Object {
        "genres": Array [
          "one",
          "two",
        ],
        "id": 1,
        "movieName": "movieName",
        "watched": false,
      }
    }
  />
  <Connect(MovieItem)
    key="1"
    movie={
      Object {
        "genres": Array [
          "three",
          "four",
        ],
        "id": 2,
        "movieName": "movieName2",
        "watched": false,
      }
    }
  />
  <Connect(MovieItem)
    key="2"
    movie={
      Object {
        "genres": Array [
          "comedy",
          "four",
        ],
        "id": 3,
        "movieName": "movieName3",
        "watched": false,
      }
    }
  />
</div>
`;
